b2share/modules/records/serializers/schemas/json.py

```diff
@@ -43,19 +47,35 @@ class DraftSchemaJSONV1(Schema):
     @pre_dump
     def filter_internal(self, data):
         """Remove internal fields from the record metadata."""
+        external_pids = []
+        bucket = None
+        try:
+            record = g.record
```

Comment:
It would be cleaner to use `if hasattr(g, 'record')` as the `AttributeError` could be raised by some other line of code too.